tests/framework/matcher/conditions.go

```diff
@@ -0,0 +1,180 @@
+package matcher
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/types"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"kubevirt.io/kubevirt/tests/framework/matcher/helper"
+)
+
+func ConditionMissingOrFalse(conditionType interface{}) types.GomegaMatcher {
+	return conditionMatcher{
+		expectedType:          conditionType,
+		expectedStatus:        k8sv1.ConditionFalse,
+		conditionCanBeMissing: true,
+	}
+}
+
+func ConditionTrue(conditionType interface{}) types.GomegaMatcher {
+	return conditionMatcher{
+		expectedType:          conditionType,
+		expectedStatus:        k8sv1.ConditionTrue,
+		conditionCanBeMissing: false,
+	}
+}
+
+type conditionMatcher struct {
+	expectedType          interface{}
+	expectedStatus        k8sv1.ConditionStatus
+	conditionCanBeMissing bool
+}
+
+func (c conditionMatcher) Match(actual interface{}) (success bool, err error) {
+	if helper.IsNil(actual) {
+		return false, nil
+	}
+
+	if helper.IsSlice(actual) {
+		// Not implemented
+		return false, nil
+	}
+
+	conditionStatus, err := getStatusForConditionType(actual, c.getExpectedType())
+	if err != nil {
+		if err == errNotFound {
+			if c.conditionCanBeMissing {
+				return true, nil
+			}
+
+		}
+		return false, nil
+	}
+
+	return conditionStatus == string(c.expectedStatus), nil
+}
+
+func (c conditionMatcher) FailureMessage(actual interface{}) string {
+	if msg := checkType(actual); msg != "" {
+		return msg
+	}
+
+	conditionStatus, err := getStatusForConditionType(actual, c.getExpectedType())
+	if err != nil {
+		if err == errNotFound {
+			conditions, _ := getConditions(actual)
+			if c.conditionCanBeMissing {
+				return ""
+			}
+
+			return fmt.Sprintf("expected condition of type '%s' was not found in %v", c.expectedType, conditions)
+		}
+
+		return err.Error()
+	}
+	conditions, _ := getConditions(actual)
+	return format.Message(conditions, fmt.Sprintf("to find condition of type '%v' and status '%s' but got '%s'", c.expectedType, c.expectedStatus, conditionStatus))
+}
+
+func (c conditionMatcher) NegatedFailureMessage(actual interface{}) string {
+	if msg := checkType(actual); msg != "" {
+		return msg
+	}
+
+	conditionStatus, err := getStatusForConditionType(actual, c.getExpectedType())
+	if err != nil {
+		if err == errNotFound {
+			conditions, _ := getConditions(actual)
+			if c.conditionCanBeMissing {
+				return fmt.Sprintf("expected condition of type '%s' was not found in %v", c.expectedType, conditions)
+			}
+		}
+		return err.Error()
+	}
+	conditions, _ := getConditions(actual)
+	return format.Message(conditions, fmt.Sprintf("to not find condition of type '%v' and status '%s' but got '%s'", c.expectedType, c.expectedStatus, conditionStatus))
+}
+
+func (c conditionMatcher) getExpectedType() string {
+	return reflect.ValueOf(c.expectedType).String()
+}
+
+func checkType(actual interface{}) string {
+	if helper.IsNil(actual) {
```

Comment:
IsNil is not the only thing to consider. Let's just use `helper.ToUnstructured` which can handle points and so forth too.